protocol/x/clob/memclob/memclob.go

```diff
@@ -85,6 +85,8 @@ func (m *MemClobPriceTimePriority) CancelOrder(
 	ctx sdk.Context,
 	msgCancelOrder *types.MsgCancelOrder,
 ) (offchainUpdates *types.OffchainUpdates, err error) {
+	lib.AssertCheckTxMode(ctx)
```

Comment:
Suggestion: note that this function should only be called in `CheckTx` / `PrepareCheckState`. Up to you